pinot-broker/src/main/java/org/apache/pinot/broker/routing/BrokerRoutingManager.java

```diff
@@ -102,11 +104,16 @@ public class BrokerRoutingManager implements RoutingManager, ClusterChangeHandle
   private final ServerRoutingStatsManager _serverRoutingStatsManager;
   private final PinotConfiguration _pinotConfig;
 
+  // Map that contains the tableNameWithType as key and the enabled serverInstances that are tagged with the table's
+  // tenant.
+  private final Map<String, Map<String, ServerInstance>> _tableTenantServersMap = new ConcurrentHashMap<>();
```

Comment:
Is it possible / worth it to invoke the new code path that works on this map only if feature flag of multi stage query engine is enabled ?

Reply:
IMO, using the condition in many places might make the code unreadable. I would prefer not having it behind the condition. Thoughts?